src/NodeParser/MappedTypeNodeParser.ts

```diff
@@ -24,23 +24,91 @@ export class MappedTypeNodeParser implements SubNodeParser {
         );
     }
 
-    private getObjectProperty(type: any, node: ts.MappedTypeNode, context: Context) {
-        return new ObjectProperty(
-            type.value,
-            this.childNodeParser.createType(node.type!, context),
-            !node.questionToken,
-        );
-    }
+    /*
+eg
+        [P in K]: T[P];
 
+typeParameter:
+        P in K
+type :
+        T[P]
+context.arguments:
+        [DefinitionType, UnionType]
+context.parameters:
+        ["T", "K"]
+     */
     private getProperties(node: ts.MappedTypeNode, context: Context): ObjectProperty[] {
-        const type: any = this.typeChecker.getTypeFromTypeNode(node.typeParameter.constraint!);
 
-        if (type.types) {
-            return type.types.map((t: any) => this.getObjectProperty(t, node, context));
-        } else if (type.intrinsicName !== "never") {
-            return [this.getObjectProperty(type, node, context)];
+        const getParameterProperties = function (typeId: string, namesOnly: boolean = false) {
+
+            // @ts-ignore
+            const t = context.arguments.find((v: any, i: any) => {
+                // @ts-ignore
+                return context.parameters[i] === typeId;
+            });
+            // @ts-ignore
+            if (t.type && t.type.properties) { // pick orig
+                // @ts-ignore
+                return t.type.properties;
+                // @ts-ignore
+            } else if (t.types) {  // pick, to pick
+                // @ts-ignore
+                return t.types.map((a: any) => a.value);
+                // @ts-ignore
+            } else if (t.properties)  { // partial orig // partial to pick
+                // @ts-ignore
+                return (namesOnly) ? t.properties.map((p: any) => p.name) : t.properties;
+            }
+
+        };
+
+        // @ts-ignore
```

Comment:
Can you remove these?

Reply:
I'd very much like to. Do you know any other way?

Reply:
Any way to do what? I'm not sure what this is supposed to fix?